livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py

```diff
@@ -245,15 +288,53 @@ async def recv_task():
                 except Exception:
                     logger.exception("failed to process AssemblyAI message")
 
-        tasks = [
-            asyncio.create_task(send_task()),
-            asyncio.create_task(recv_task()),
-        ]
+        async def _wait_for_reconnect():
+            await self._reconnect_event.wait()
+
+        while True:
+            try:
+                ws = await self._connect_ws()
+                tasks = [
+                    asyncio.create_task(send_task(ws)),
+                    asyncio.create_task(recv_task(ws)),
+                ]
+                reconnect_task = asyncio.create_task(_wait_for_reconnect())
```

Comment:
This should work
```suggestion
                reconnect_task = asyncio.create_task(self._reconnect_event.wait())
```

Reply:
That one will give type error, Value of type variable "_FT" of "wait" cannot be "object"

Reply:
It'll inside asyncio.wait, but not inside asyncio.create_task no?

Reply:
Yes

Reply:
I've updated code with ignoring types, let me know what you think?